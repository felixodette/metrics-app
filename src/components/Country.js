/* eslint-disable no-unused-vars */
import React from 'react';
import PropTypes from 'prop-types';
import styles from './Country.module.css';

function Country(props) {
  const { name, number, vector } = props;
  const map = { background: `url(${vector}) no-repeat center center/cover` };

  return (
    <div className={styles.wrapper}>
      <i className={styles.right_arrow} />
      <i style={map} className={styles.map} />
      <div className={styles.info}>
        <h2>{name}</h2>
        <span>{number}</span>
      </div>
    </div>
  );
}

export default Country;

Country.propTypes = {
  name: PropTypes.string.isRequired,
  number: PropTypes.number.isRequired,
  vector: PropTypes.string.isRequired,
};
