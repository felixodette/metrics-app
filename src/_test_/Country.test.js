import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/configureStore';
import Country from '../components/Country';

describe('Country component test', () => {
  it('Renders Country', () => {
    const country = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Country name="test" number={5} vector="null" cities={[]} />
          </Router>
        </Provider>,
      )
      .toJSON();
    expect(country).toMatchSnapshot();
  });
});
