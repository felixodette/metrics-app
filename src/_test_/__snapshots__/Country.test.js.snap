// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country component test Renders Country 1`] = `
<a
  className="wrapper"
  href="/cities"
  onClick={[Function]}
>
  <i
    className="right_arrow"
  />
  <i
    className="map"
    style={
      Object {
        "background": "url(null) no-repeat center center/cover",
      }
    }
  />
  <div
    className="info"
  >
    <h3>
      test
    </h3>
    <span>
      5
    </span>
  </div>
</a>
`;
